cmd/maya-exporter/app/collector/pool/pool.go

```diff
@@ -0,0 +1,172 @@
+package pool
+
+import (
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/util"
+	zpool "github.com/openebs/maya/pkg/zpool/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// pool implements prometheus.Collector interface
+type pool struct {
+	sync.Mutex
+	metrics
+	request bool
+}
+
+var (
+	// runner variable is used for executing binaries
+	runner util.Runner
+)
+
+// InitVar initialize runner variable
+func InitVar() {
+	runner = util.RealRunner{}
+}
+
+// New returns new instance of pool
+func New() *pool {
+	return &pool{
+		metrics: newMetrics(),
+	}
+}
+
+func (p *pool) isRequestInProgress() bool {
+	return p.request
+}
+
+func (p *pool) setRequestToFalse() {
+	p.Lock()
+	p.request = false
+	p.Unlock()
+}
+
+// GetInitStatus run zpool binary to verify whether zpool container
+// has started.
+func (p *pool) GetInitStatus(timeout time.Duration) {
+
+	for {
+		stdout, err := zpool.Run(timeout, runner, "status")
+		if err != nil {
+			glog.Warningf("Failed to get zpool status, error: %v, pool container may be initializing,  retry after 2s", err)
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		str := string(stdout)
+		if zpool.IsNotAvailable(str) {
+			glog.Warning("No pool available, pool must be creating, retry after 3s")
+			time.Sleep(3 * time.Second)
+			continue
+		}
+		glog.Info("\n", string(stdout))
+		break
+	}
+}
+
+// collectors returns the list of the collectors
+func (p *pool) collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		p.size,
+		p.status,
+		p.usedCapacity,
+		p.freeCapacity,
+		p.usedCapacityPercent,
+		p.zpoolCommandErrorCounter,
+		p.zpoolRejectRequestCounter,
+		p.zpoolListparseErrorCounter,
+		p.noPoolAvailableErrorCounter,
```

Comment:
should we add collector for `incompleteOutputErrorCounter` also?

Reply:
yes, thanks for pointing this.